### PR TITLE
#2433 Issues with Saving and Opening Structures with Multi-Tailed Arrow in CDX and CDXML Formats

### PR DESCRIPTION
**Pull request by even1024**
## Generic request
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name does not contain '#'
- [x] base branch (master or release/xx) is correct
- [x] PR is linked with the issue
- [x] task status changed to "Code review"
- [x] code follows product standards
- [x] regression tests updated

### CODE REVIEW

File: core/indigo-core/molecule/src/base_molecule.cpp [hunk 1]
```diff
@@ -4094,6 +4094,39 @@ void BaseMolecule::getAtomsCenterPoint(Array<int>& atoms, Vec2f& res)
...
+    double bondSum = 0.0;
+    for (auto j : edges())
+    {
+        const Edge& edge = getEdge(j);
+        auto& v1 = getAtomXyz(edge.beg);
+        auto& v2 = getAtomXyz(edge.end);
+        float bondLength = std::hypot(v1.x - v2.x, v1.y - v2.y);
+        bondSum += bondLength;
+    }
+    return static_cast<float>(bondSum / edgeCount());
```
Comment (AlexanderSavelyev) [inline core/indigo-core/molecule/src/base_molecule.cpp:4116]: add edgeCount=0 check

Comment (even1024) [reply core/indigo-core/molecule/src/base_molecule.cpp]: fixed

File: core/indigo-core/layout/pathway_layout.h [hunk 2]
```diff
@@ -73,8 +75,18 @@ namespace indigo
                 if (reactantIdx != -1)
                 {
                     auto& mol = reaction.getMolecule(reactantIdx);
-                    // MoleculeLayout molLayout(mol, true);
-                    // molLayout.make();
+                    auto mean = mol.getBondsMeanLength();
+                    if (mean < MIN_BOND_MEAN)
```
Comment (AliaksandrDziarkach) [inline core/indigo-core/layout/pathway_layout.h:79]: Maybe add check for bond count > 0?

File: core/indigo-core/layout/pathway_layout.h [hunk 3]
```diff
@@ -86,8 +98,19 @@ namespace indigo
                     for (auto pidx : simpleReaction.productIndexes)
                     {
                         auto& mol = reaction.getMolecule(pidx);
-                        // MoleculeLayout molLayout(mol, true);
-                        // molLayout.make();
+                        auto mean = mol.getBondsMeanLength();
+                        if (mean < MIN_BOND_MEAN)
```
Comment (AliaksandrDziarkach) [inline core/indigo-core/layout/pathway_layout.h:102]: same here

File: core/indigo-core/layout/pathway_layout.h (hunk 2 above)
Comment (even1024) [reply core/indigo-core/layout/pathway_layout.h:79]: There is no reason for this check. Let's layout will decide what to do if bond count > 0 or not.

**Review by AliaksandrDziarkach — APPROVED** [review]

**Pull request merged**